sos/plugins/kubernetes.py

```diff
@@ -69,7 +69,9 @@ def setup(self):
 
         # get all namespaces in use
         kn = self.get_command_output('%s get namespaces' % kube_cmd)
-        knsps = [n.split()[0] for n in kn['output'].splitlines()[1:] if n]
+        # namespace is the 1st word on line, until the line has spaces only
+        knsps = [n.split()[0] for n in kn['output'].splitlines()[1:]
+                 if n and len(n.split())]
```

Comment:
I wonder if this might be more readable as:

```python
    kn_output = kn['output'].splitlines()[1:]
    knsps = [n.split()[0] for n in kn_output if n and len(n.split())]
```

70 chars on the long line... I don't know, but that hanging `if` really confused my eyes for a moment on the first look.

Reply:
Good point for more code clarity there, updating the PR..